packages/ddp-server/crossbar.js

```diff
@@ -63,8 +66,10 @@ _.extend(DDPServer._Crossbar.prototype, {
             self.factPackage, self.factName, -1);
         }
         delete self.listenersByCollection[collection][id];
-        if (_.isEmpty(self.listenersByCollection[collection])) {
+        self.listenersByCollectionCount[collection]--;
+        if (self.listenersByCollectionCount[collection] === 0) {
           delete self.listenersByCollection[collection];
+          delete self.listenersByCollectionCount[collection];
```

Comment:
Couldn't this just stay 0, instead of needing to be deleted?

Reply:
yep, I'm fine w/ either! Just kinda nice if `listenersByCollection` and `listenersByCollectionCount` always have the same keys.